### PR TITLE
[AVG-373] Fix authorization token format

### PR DESCRIPTION
**Pull request by pabloschz**
Make authorization token works and force SSL only when PRO env is detected

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,10 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## [5.13.4] - 2021-05-03
```
Comment (carlosescri) [inline CHANGELOG.md:7]:
```suggestion
## [5.13.4] - 2022-05-03
```

**Review by carlosescri — CHANGES_REQUESTED** [review]
Please, change `.coffee` files instead of `.js` files (those are generated during compilation).

**Comment by pabloschz** [comment]
> Please, change `.coffee` files instead of `.js` files (those are generated during compilation).

Woups, changed thanks!

File: src/client.coffee [hunk 2]
```diff
@@ -77,8 +78,7 @@ class Client
     if secret?
       @requestOptions.headers["Authorization"] = secret
 
-    # This works even if no apiKey passed but passed an "Authorization" header
-    if "Authorization" of @requestOptions.headers
+    if address.search(/doofinder.com/) != -1
```
Comment (carlosescri) [inline src/client.coffee:81]: Why this change?

Comment (pabloschz) [reply src/client.coffee:81]: I need to disable https and SSL security in local development because with this it didin't work.
So the unique way I found is to check if the doofinder.com is in the target address, check if the authentication token is defined is not longer valid as we are gonna use it in local enviroment also and playground for example wouldn't work.

Comment (carlosescri) [reply src/client.coffee:81]: 🤔 maybe this could help… starting refactoring at line 75…

```coffeescript
if secret?
  @requestOptions.headers["Authorization"] = secret

if protocol?
  @requestOptions.protocol = "#{protocol}:"
elif "Authorization" of @requestOptions.headers
  @requestOptions.protocol = "https:"
  forceSSL = true
```

I haven't tried it but I think that this would allow you to force the protocol even providing the auth token.

**Review by carlosescri — APPROVED** [review]

**Comment by carlosescri** [comment]
Ahora te queda pegarte con los tests…

**Review by carlosescri — APPROVED** [review]
